### PR TITLE
Report connection error status code

### PR DESCRIPTION
**Pull request by jasongin**
When a websocket connection to the relay fails, check the HTTP status code and use that to generate a better error message in the `TunnelConnectionException`. That exception message will also be displayed by the CLI.

### CODE REVIEW

**Comment by jasongin** [comment]
This needs a rebase...

File: cs/src/Connections/WebSocketStream.cs [hunk 1]
```diff
@@ -6,519 +6,547 @@
...
             {
-                configure?.Invoke(socket.Options);
-                await socket.ConnectAsync(uri, cancellation);
-                return new WebSocketStream(socket);
-            }
-            catch
-            {
-                socket.Dispose();
-                throw;
+                var offset = messageIndex + statusCodeMessage.Length;
```
Comment (IlyaBiryukov) [inline cs/src/Connections/WebSocketStream.cs:110]: There is already code in `WebSocketStream` that gets the status code from the exception message, checked in with my PR https://github.com/microsoft/dev-tunnels/pull/99

Comment (jasongin) [reply cs/src/Connections/WebSocketStream.cs:110]: Yes, I just realized this is mostly duplicating that. I'll close this PR.

File: cs/src/Connections/WebSocketStream.cs [hunk 2]
```diff
@@ -6,519 +6,547 @@
 using System;
 using System.Diagnostics;
 using System.IO;
+using System.Net;
 using System.Net.WebSockets;
 using System.Threading;
 using System.Threading.Tasks;
 
-namespace Microsoft.VsSaaS.TunnelService
+namespace Microsoft.VsSaaS.TunnelService;
```
Comment (IlyaBiryukov) [inline cs/src/Connections/WebSocketStream.cs:14]: Lots of changes in this file making it hard to review. Perhaps changing to file-scoped namespace may happen in a separate PR to simplify reviewing this one?

Comment (jasongin) [reply cs/src/Connections/WebSocketStream.cs:14]: For future reference, GitHub can hide whitespace differences:
image

But I closed this PR anyway.

Comment (IlyaBiryukov) [reply cs/src/Connections/WebSocketStream.cs:14]: Thanks for the tip on the whitespace diffing!

**Pull request closed without merging**